src/tiled/objectselectiontool.cpp

```diff
@@ -825,6 +825,44 @@ void ObjectSelectionTool::updateHandleVisibility()
     for (int i = 0; i < AnchorCount; ++i)
         mResizeHandles[i]->setVisible(showHandles && mMode == Resize);
 
+    if (hasSelection) {
+        int selectedObjectsCount = mapDocument()->selectedObjects().count();
+        MapObject *firstSelectedObj =mapDocument()->selectedObjects().at(0);
+        qreal totalHeight = 0.0, totalWidth = 0.0;
+        foreach (MapObject *mapObj, mapDocument()->selectedObjects()) {
+            if (!mapObj->polygon().isEmpty()) {
+                totalHeight += mapObj->polygon().boundingRect().height();
+                totalWidth += mapObj->polygon().boundingRect().width();
+            } else {
+                totalHeight += mapObj->bounds().height();
+                totalWidth += mapObj->bounds().width();
+            }
+        }
+
+
+        /*
+         * Check whether we haven't selected single rectangle, ellipse or tile or we have selected
+         * multiply objects.
+         */
+        if ((selectedObjectsCount == 1 &&
+                firstSelectedObj->shape() == MapObject::Polyline &&
+                firstSelectedObj->polygon().first().y() == firstSelectedObj->polygon().last().y()) ||
```

Comment:
This check on the y coordinate is wrong. You should check the bounding box, since you may have a polyline that has for example three points, with the first and the last on the same y but the middle node lower or higher.

Also, a `MapObject::Polygon` can have 0 width or height just as well and can't be resized then either, so it's wrong to only check for polylines.